packages/react/src/transactions/infrastructure/__tests__/TransactionFactory.spec.ts

```diff
@@ -373,148 +373,129 @@ describe(`Given an instance of the ${TransactionFactory.name}`, () => {
+          status: ProxyActionStatus.MINTING,
         });
+
+        const proxyActionStatusEvent = mockProxyActionStatusEvent({
+          txHash: mockTransactionHash(),
+          status: ProxyActionStatus.TRANSFERRING,
+        });
+
+        const observer = MockedTransactionObserver.withIndexingEventsSequence({
+          indexingId: init.proxyId,
+          indexingEventsSequence: [proxyActionStatusEvent],
+        });
+        const factory = setupTransactionFactory({ observer });
+
+        const transaction = factory.createProxyTransaction(init);
+        const result = await transaction.waitNextEvent();
+
+        expect(result.unwrap()).toBe(TransactionEvent.BROADCASTED);
+        expect(transaction.hash).toEqual(proxyActionStatusEvent.txHash);
+        expect(transaction.status).toBe(ProxyActionStatus.TRANSFERRING);
+      });
+
+      it(`should reflect tx hash changes  the proxy action status is ${ProxyActionStatus.MINTING}`, async () => {
+        const init = mockProxyTransactionData<SupportedTransactionRequest>({
+          status: ProxyActionStatus.MINTING,
+        });
+
+        const proxyActionStatusEvent = mockProxyActionStatusEvent({
+          txHash: mockTransactionHash(),
+          status: ProxyActionStatus.MINTING,
+        });
+
+        const observer = MockedTransactionObserver.withIndexingEventsSequence({
+          indexingId: init.proxyId,
+          indexingEventsSequence: [proxyActionStatusEvent],
+        });
+        const factory = setupTransactionFactory({ observer });
+
+        const transaction = factory.createProxyTransaction(init);
+        const result = await transaction.waitNextEvent();
+
+        expect(result.unwrap()).toBe(TransactionEvent.UPGRADED);
+        expect(transaction.hash).toEqual(proxyActionStatusEvent.txHash);
+        expect(transaction.status).toBe(ProxyActionStatus.MINTING);
+      });
+
+      it(`should reflect tx hash changes  the proxy action status is ${ProxyActionStatus.TRANSFERRING}`, async () => {
+        const init = mockProxyTransactionData<SupportedTransactionRequest>({
+          status: ProxyActionStatus.TRANSFERRING,
+        });
+
+        const proxyActionStatusEvent = mockProxyActionStatusEvent({
+          txHash: mockTransactionHash(),
+          status: ProxyActionStatus.TRANSFERRING,
+        });
+
+        const observer = MockedTransactionObserver.withIndexingEventsSequence({
+          indexingId: init.proxyId,
+          indexingEventsSequence: [proxyActionStatusEvent],
+        });
+        const factory = setupTransactionFactory({ observer });
+
+        const transaction = factory.createProxyTransaction(init);
+        const result = await transaction.waitNextEvent();
+
+        expect(result.unwrap()).toBe(TransactionEvent.UPGRADED);
+        expect(transaction.hash).toEqual(proxyActionStatusEvent.txHash);
+        expect(transaction.status).toBe(ProxyActionStatus.TRANSFERRING);
```

Comment:
Refactored these tests cause it was very unclear and confusing and from the test alone I couldn't figure out if the relevant logic was affected by the changes I need to do or not.